### DISCUSSION TITLE
FIX to Json and Json to FIX Transformation

### DISCUSSION THREAD

**Discussion by rajendra1** [quickfix-j/quickfixj#412] [Ideas]
I see Message.toXML converts the FIX message to XML,  why don't have similar approach for JSON as well ?

I designing a Solution which will bridge the Kafka Cluster with Fix Engine, I was initially looking for Fix transformation library with  Fix transformation  [link](https://jcustenborder.github.io/kafka-connect-documentation/projects/kafka-connect-transform-fix/transformations/examples/FromFIX.example001.html) Connector but had  to drop idea for other reasons.

can we have such transformer API library within core QuickFixJ ? 
My requirement is to transform JSON to FIX and vis a versa .
 
 I wrote my own parser for  Fix.4.2  which converts Fix to Json, but I feel the need of efficiency and validation with all message type.

Rajendra Patil
https://www.linkedin.com/in/rajendra-patil/

**Comment by chrjohn** [comment]
I don't see why there shouldn't be a `toJson()` method. However, up to now no one has asked for this feature. I assume the users who need it implement it by themselves. We welcome pull requests so if you want to give it a shot, feel free to open a PR.
Thanks,
Chris.

**Reply by rajendra1** [reply]
Hi Chris, 
        You are absolutely right, as I also wrote my own, but what I notice, if mapping Data dictionary explicitly for a message if done in Constructor of quickfix.Message instance while it gets to know the Version, it will be simpler.
Well as you said its  good idea, to raise a Pull request, will see if I could  get changes to spare time on it as its little different approach altogether.

By the way I wrote one more method in our custom transformer , i.e.  **jsonToFix**(jsonString, [model extends Object].class)  this helps us mapping any Json to FIX.

**Comment by stephennimmo** [comment]
One time I tried to write a library that would convert the fix dictionary xml to a set of protobuf definitions. The numbering of the protobuf fields worked out great until I got to the repeating groups and that's when the model broke down for me.

**Comment by david-gibbs-ig** [comment]
I'm building JSON schema from a FIX Orchestration, see https://github.com/IG-Group/ig-orchestrations/tree/master/ig-us-rfed. This is based on work by the FIX Trading Community. It is not 100% but it works, at least up to a point. In due course I expect to see the re-emergence of a FIX/JSON working group. I'm not sure personally if this should be part of the **core/fundamental** QuickFIX/J repo. I also think it can be better for it to not be subject to the QuickFIXJ license. I do think that a QFJ integration would be great though, there are a lot of ways to do it I think.

Please have a look at https://github.com/quickfix-j/quickfixj/pull/460, work on integrating FIX Latest is coming together and this has leaned heavily towards making customisation practical.

Please note there there are also tools for orchestra to protobuf and to SBE. These are at varying levels of maturity but I think the idea is sound.

**Comment by hasansjc** [comment]
Hey there, I want to convert the JSON message to FIX format. I have converted FIX to JSON by using this code https://gist.github.com/hstaos/3711a7f55eecb63db52c9855967a336d  But I don't have any clue how to do it. What are the prerequisite for doing that. So can anyone please guide me how to achieve this. Thank you
